---
layout: page
title: "Systematic mapping of chromatin state landscapes during mouse development"
breadcrumb: true
categories:
    - preprint
## publication related information
pub:
    authors: " David Gorkin*, Iros Barozzi*, Yanxiao Zhang*, Ah Young Lee, Bin Lee, Yuan Zhao, Andre Wildberg, Bo Ding, Bo Zhang, Mengchi Wang, J Seth Strattan, Jean M Davidson, Yunjiang Qiu, Veena Afzal, Jennifer A Akiyama, Ingrid Plajzer-Frick, Catherine S Pickle, Momoe Kato, Tyler H Garvin, Quan T Pham, Anne N Harrington, Brandon J Mannion, Elizabeth A Lee, Yoko Fukuda-Yuzawa, Yupeng He, Sebastian Preissl, Sora Chee, Brian A Williams, Diane Trout, Henry Amrhein, Hongbo Yang, J Michael Cherry, Yin Shen, Joseph R Ecker, Wei Wang, Diane E Dickel, Axel Visel, Len A Pennacchio, Bing Ren"
    journal: " bioRxiv"
    date: 2017-07-21
    doi:  https://doi.org/10.1101/166652
    abstract: "Embryogenesis requires epigenetic information that allows each cell to respond appropriately to developmental cues. Histone modifications are core components of a cells epigenome, giving rise to chromatin states that modulate genome function. Here, we systematically profile histone modifications in a diverse panel of mouse tissues at 8 developmental stages from 10.5 days post conception until birth, performing a total of 1,128 ChIP-seq assays across 72 distinct tissue-stages. We combine these histone modification profiles into a unified set of chromatin state annotations, and track their activity across developmental time and space. Through integrative analysis we identify dynamic enhancers, reveal key transcriptional regulators, and characterize the role of chromatin-based repression in developmental gene regulation. We also leverage these data to link enhancers to putative target genes, revealing connections between coding and non-coding sequence variation in disease etiology. Our study provides a compendium of resources for biomedical researchers, and achieves the most comprehensive view of embryonic chromatin states to date."

---
