---
layout: page
title: "MAPS: model-based analysis of long-range chromatin interactions from PLAC-seq and HiChIP experiments"
breadcrumb: true
categories:
    - preprint
## publication related information
pub:
    authors: "Ivan Juric, Miao Yu, Armen Abnousi, Ramya Raviram, Rongxin Fang, Yuan Zhao, Yanxiao Zhang, Yuchen Yang, Yun Li, Bing Ren, Ming Hu"
    journal: " bioRxiv"
    date: 2017-11-14
    doi:  https://doi.org/10.1101/411835
    abstract: "Dynamic DNA methylation and three-dimensional chromatin architecture compose a major portion of the epigenome and play an essential role in tissue specific gene expression programs. Currently, DNA methylation and chromatin organization are generally profiled in separate assays. Here, we report Methyl-HiC, a method combining in situ Hi-C and whole genome bisulfite sequencing (WGBS) to simultaneously capture chromosome conformation and DNA methylome in a single assay. Methyl-HiC analysis of mouse embryonic stem cells reveals coordinated DNA methylation between distant yet spatially proximal genomic regions. Extension of Methyl-HiC to single cells further enables delineation of the heterogeneity of both chromosomal conformation and DNA methylation in a mixed cell population, and uncovers increased dynamics of chromatin contacts and decreased stochasticity in DNA methylation in genomic regions that replicate early during cell cycle."
---
